sycamore/evaluation/pipeline.py

```diff
@@ -1,116 +1,143 @@
-                "ext": {
-                    "generative_qa_parameters": {
-                        "llm_question": doc["question"],
-                        "context_size": self._os_config["context_window"],
-                        "llm_model": self._os_config["llm"],
-                    }
-                },
-                "size": self._os_config["size"],
-            }
-        else:
-            query["query"] = {
-                "_source": {"excludes": ["embedding"]},
-                "query": {
-                    "bool": {
-                        "must": [
-                            {
-                                "hybrid": {
-                                    "queries": [
-                                        {"match": {"text_representation": doc["question"]}},
-                                        {
-                                            "neural": {
-                                                "embedding": {
-                                                    "query_text": doc["question"],
-                                                    "model_id": self._os_config["embedding_model_id"],
-                                                    "k": self._os_config["neural_search_k"],
-                                                }
-                                            }
-                                        },
-                                    ]
+        query["query"] = {
+            "_source": {"excludes": ["embedding"]},
+            "query": {
+                "hybrid": {
+                    "queries": [
+                        {"match": {"text_representation": doc["question"]}},
+                        {
+                            "neural": {
+                                "embedding": {
+                                    "query_text": doc["question"],
+                                    "model_id": self._os_config["embedding_model_id"],
+                                    "k": self._os_config["neural_search_k"],
+                                    "filter": [{"exists": {"field": "parent_id"}}],
                                 }
                             }
-                        ],
-                        "filter": [{"exists": {"field": "parent_id"}}],
-                    }
+                        },
+                    ]
+                }
+            },
+            "size": self._os_config["size"],
+        }
+
+        if "llm" in self._os_config:
+            query["params"] = {"search_pipeline": self._os_config["search_pipeline"]}
+            query["query"]["ext"] = {
+                "generative_qa_parameters": {
+                    "llm_question": doc["question"],
+                    "context_size": self._os_config["context_window"],
+                    "llm_model": self._os_config["llm"],
                 },
-                "size": self._os_config["size"],
+                "rerank": {"query_context": {"query_text": doc["question"]}},
```

Comment:
Oh we're assuming rerank then?

Reply:
Yeah this is basically our "default" os query. I'm sure we'll have cases where we'll want to customize this